contracts/double-dice-roll/src/contract.rs

```diff
@@ -100,6 +103,9 @@ pub fn execute_roll_dice_multiple_times(
     let nois_proxy = NOIS_PROXY.load(deps.storage)?;
 
     let mut msgs = Vec::<WasmMsg>::new();
+    if job_id.len() + 2 + n_times.to_string().len() > MAX_JOB_ID_LEN {
+        return Err(ContractError::JobIdTooLong);
+    }
```

Comment:
I'd move this in the loop and check the composed job ID directly. Otherwise this is hard to maintain and audit.